### DISCUSSION TITLE
jpype issue on Windows with non-ascii paths

### DISCUSSION THREAD

**Discussion by villares** [tabreturn/thonny-py5mode#56] [Ideas]
I had recently other "international" students with non-ascii usernames on Windows that struggled to make py5 work, they had no idea of what was going on, as the message just communicates the JVM coudn't be started... I was lucky to remember this might have been the issue. I wonder how many people silently give up on it :(

https://github.com/jpype-project/jpype/issues/1111

Perhaps we could we add a warning message on the plug-in if it detects the JAVA_HOME path contains non-ascii chars in order to mitigate this issue while it is not fixed?

**Comment by hx2A** [comment]
What do I need to do to reproduce this? If I create a Windows directory with some non-ascii characters, put the JVM in a subdirectory, set JAVA_HOME, I would not be able to start jpype? Can I reproduce this on my Windows machine?

> Perhaps we could we add a warning message on the plug-in if it detects the JAVA_HOME path contains non-ascii chars in order to mitigate this issue while it is not fixed?

This warning should be in py5, not the plugin. Someone might be using py5 outside of Thonny. Those users are also exposed to this problem.

I would rather take a stab at fixing it though. What Thrameos wrote in that thread might be enough to get me started.

**Reply by villares** [reply]
Cool, yes, putting the JVM on a subdir with non-ascii characters in the path (example: `C:/molnár/...`), note the `á`, should let you reproduce it. Thanks!

**Reply by hx2A** [reply]
I'll give it a try then!

**Reply by villares** [reply]
@hx2A this has been fixed at `jpype`'s end, right? Has it been released by them? I will add checking this on Windows to my list TODO list ...

Update: Not that fast. They had no new releases since December 2023 :cry: 
https://github.com/jpype-project/jpype/issues/1111#issuecomment-2072835916

**Reply by hx2A** [reply]
> Update: Not that fast. They had no new releases since December 2023

Right, the latest version, 1.5.0 I believe, has been out for months. I just looked at that repo to see if a release was coming soon and I didn't see anything. Typically they have an open issue with a checklist you can follow to know how soon the release will take place.

**Comment by villares** [comment]
This was fixed! 🥳 🎉 ✨
